ci-operator/step-registry/cucushift/upgrade/hypershift/cp/toimage/cucushift-upgrade-hypershift-cp-toimage-commands.sh

```diff
@@ -0,0 +1,232 @@
+    if ${OC} get clusteroperator -o json | jq '.items[].status.conditions[] | select(.type == "Degraded") | .status'  | grep -iv 'False'; then
+        echo >&2 "Some operators are Degraded, pls run 'oc get clusteroperator -o json' to check"
+        (( tmp_ret += 1 ))
+    fi
+
+    return $tmp_ret
+}
+
+function wait_clusteroperators_continous_success() {
+    local try=0 continous_successful_check=0 passed_criteria=3 max_retries=20
+    while (( try < max_retries && continous_successful_check < passed_criteria )); do
+        echo "Checking #${try}"
+        if check_clusteroperators; then
+            echo "Passed #${continous_successful_check}"
+            (( continous_successful_check += 1 ))
+        else
+            echo "cluster operators are not ready yet, wait and retry..."
+            continous_successful_check=0
+        fi
+        sleep 60
+        (( try += 1 ))
+    done
+    if (( continous_successful_check != passed_criteria )); then
+        echo >&2 "Some cluster operator does not get ready or not stable"
+        echo "Debug: current CO output is:"
+        oc get co
+        return 1
+    else
+        echo "All cluster operators status check PASSED"
+        return 0
+    fi
+}
+
+function health_check() {
+    #1. Check all cluster operators get stable and ready
+    echo "Step #1: check all cluster operators get stable and ready"
+    wait_clusteroperators_continous_success
+
+    #2. Make sure every machine is in 'Ready' status
+    echo "Step #2: Make sure every machine is in 'Ready' status"
+    check_node
+
+    #3. All pods are in status running or complete
+    echo "Step #3: check all pods are in status running or complete"
+    check_pod
+}
+
+if [ ! -f "${SHARED_DIR}/mgmt_kubeconfig" ]; then
+  exit 1
+fi
+echo "switch kubeconfig"
+export KUBECONFIG="${SHARED_DIR}/mgmt_kubeconfig"
+
+# Setup proxy if it's present in the shared dir
+if [[ -f "${SHARED_DIR}/proxy-conf.sh" ]]; then
+    # shellcheck disable=SC1091
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+export OC="run_command_oc"
+
+echo "OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE: $OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE"
+cluster_name=$(oc get hostedclusters -n "$HYPERSHIFT_NAMESPACE" -o jsonpath='{.items[0].metadata.name}')
+oc patch hostedcluster "$cluster_name" -n "$HYPERSHIFT_NAMESPACE" --type=merge -p '{"spec":{"release":{"image":"'"${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"'"}}}'
+
+_upgradeReady=0
+for ((i=1; i<=60; i++)); do
+  count=$(oc get hostedcluster -n clusters "$cluster_name" -ojsonpath='{.status.version.history[?(@.image=="'"${OPENSHIFT_UPGRADE_RELEASE_IMAGE_OVERRIDE}"'")].state}' | grep -c Completed || true)
+  if [ "$count" -eq 1 ] ; then
+      echo "HyperShift HostedCluster(CP) upgrade successful"
+      _upgradeReady=1
+      break
+  fi
+  echo "Try ${i}/60: HyperShift HostedCluster(CP) is not updated yet. Checking again in 30 seconds"
+  sleep 30
+done
+
+if [ $_upgradeReady -eq 0 ]; then
```

Comment:
In shell, 0 generally means succeeded status. Shall we revert the logic to keep it consistent ?

Reply:
@liangxia Done, Thanks for your comment